wlroots-sys/src/lib.rs

```diff
@@ -35,6 +35,26 @@ mod generated {
                 include!(concat!(env!("OUT_DIR"), "/server_decoration_server_api.rs"));
             }
         }
+        pub mod idle {
```

Comment:
I mainly copied-pasted this part and subbed wl_surface with wl_seat. What's the point of this section?

Reply:
The code generated in the build script is placed into files, specifically `<interface_name>_api.rs` and `<interface_name>_interfaces.rs`. We need these definitions because they implement glue code for taking our Rust types and marshaling them into a form to be sent over the Wayland socket. Since we marshal to this type, this means the response can be consumed by any language (because they can convert from the Wayland types to their lang types).

The use statements are here because we need literally copy and paste the code from the files (kind of like how `#include` works in C actually) using the `include!` macro.

It needs to be in a module because this generates a lot of code we don't care about. The only code we about is in the generated server module, which we re-export at the end of the file.

Reply:
So considering that I only did it for idle, should I do a PR for the rest of the extensions that I've done so far? Since from what I understand, the other extensions aren't **still** linked up properly then..

Reply:
Hm, actually looking at this again I don't think this is necessary. I haven't looked at the extensions in a while, so it might have changed.

This should not be here actually, since wlroots already links to it. I believe this to be the case, so this generation code can actually be removed since it's dead code.

Reply:
Yeah this code is dead. The only one that is needed is the server decoration, and that's for an enum defined in the protocol. So sorry about that, your other protocols are fine. So is this one, it's just that this code is unnecessary. I'll make an issue for that now: https://github.com/swaywm/wlroots-rs/issues/266